packages/mwp-app-render/src/components/connectedWithMatchMedia.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createStructuredSelector, createSelector } from 'reselect';
+import withMatchMedia from 'meetup-web-components/lib/utils/components/withMatchMedia';
+
+/**
+ * getConfig
+ * @param  {Object} state
+ * @return {Object} the config from state
+ */
+const getConfig = state => state.config || {};
+
+/**
+ * Checks if object is empty
+ * @param  {Object} object
+ * @return {Boolean} whether object is empty
+ */
+const isEmpty = object => object === undefined || Object.keys(object).length === 0;
+
+/**
+ * @param {Object} device - object containing device information
+ * @returns {Object} - default media object for `withMatchMedia`
+ */
+const getMediaOrDefault = (media, device) => {
+	if (!isEmpty(media)) {
+		return media;
+	}
+	if ( !isEmpty(device) ) {
+		const {isMobile, isTablet, isDesktop} = device;
+		const isTabletOrDesktop = isTablet || isDesktop;
+		return {
+			isAtSmallUp: Boolean(isMobile || isTabletOrDesktop),
+			isAtMediumUp: Boolean(isTabletOrDesktop),
+			isAtLargeUp: Boolean(isDesktop),
+		}
+	}
+	return {};
+};
+
+const mapStateToProps = createStructuredSelector({
+	device: createSelector(getConfig, config => (config.device || {})),
+});
+
+/**
+ * connectedWithMatchMediaComponent
+ * HOC that provides a default based on `device` in state.config
+ * if withMatchMedia has not yet detected the media
+ *
+ * `device` prop is provided from redux state in config.
+ * `media` prop is provided to the wrapped component in render().
+ *
+ * @returns {React.element}
+ */
+const connectedWithMatchMedia = WrappedComponent => {
+
+	/**
+	 * @module ConnectedWithMatchMedia
+	 */
+
+	class ConnectedWithMatchMedia extends React.Component {
+
+		render() {
+			const {device, media: currentMedia } = this.props;
+			const media = getMediaOrDefault(currentMedia, device);
+			return <WrappedComponent {...this.props} media={media}/>;
+		}
+	};
+	const wrappedComponentName =
+		WrappedComponent.displayName || WrappedComponent.name || 'Component';
+
+	ConnectedWithMatchMedia.WrappedComponent = WrappedComponent;
```

Comment:
What is this class prop used for?

Reply:
I'm not using it. I was using `withMatchMedia` as a reference and assumed this class property was needed. Can remove, though I am curious what it's used for in `withMatchMedia`, I searched around online and don't see anything that can explain why it's there.

Reply:
I think `connect` from react-redux also assigns it, presumably in order to maintain a reference to the wrapped component for debugging and such, which is fine. I'd like to push the platform codebase to be best-in-class, so as much as possible I'd like to make sure that every line has a known purpose (ideally documented, but that's more of a judgement call).

Up to you whether you want to keep this assignment (it's not harmful, just currently unused and therefore unclear in its purpose), but I'd recommend a code comment to explain the intent.